pkg/registry/core/pod/rest/subresources.go

```diff
@@ -48,7 +48,7 @@ var proxyMethods = []string{"GET", "POST", "PUT", "PATCH", "DELETE", "HEAD", "OP
 
 // New returns an empty pod resource
 func (r *ProxyREST) New() runtime.Object {
-	return &api.Pod{}
+	return &api.PodProxyOptions{}
```

Comment:
Given that the New method is not fully typed we should make sure none of the callers of Storage::New() are depending on one of these types.

Reply:
The only caller of Storage::New() is https://github.com/kubernetes/kubernetes/blob/c2536e2b0d6be62253815fca701736b04a9423f8/staging/src/k8s.io/apiserver/pkg/endpoints/installer.go#L145, which is then used to determined the gvk. For Connector, there are two uses of the gvk:
1. generating API doc, that results in the openapi/swagger changes in this PR.
2. assigned to reqScope.Kind, which is then passed to admission controllers, so I searched the uses of attribute.GetKind() in all admission controllers in kubernetes apiserver and in the generic apiserver. The only place the kind change matters is the admission webhook controller, which uses the kind to determine how to convert the object, which is intended.